src/accel/GeantStepDiagnostic.hh

```diff
@@ -0,0 +1,70 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/GeantStepDiagnostic.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <G4Track.hh>
+
+#include "corecel/Types.hh"
+#include "corecel/io/OutputInterface.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Tally the steps per track transported with Geant4 for each particle type.
+ *
+ * For the diagnostic class that collects the same result for tracks
+ * transported with Celeritas, see: \sa StepDiagnostic.
+ */
+class GeantStepDiagnostic final : public OutputInterface
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using VecCount = std::vector<size_type>;
+    using VecVecCount = std::vector<VecCount>;
+    using MapIntVecCount = std::unordered_map<int, VecCount>;
+    //!@}
+
+  public:
+    // Construct with number of bins and threads
+    GeantStepDiagnostic(size_type num_bins, size_type num_threads);
+
+    //! Default destructor
+    ~GeantStepDiagnostic();
```

Comment:
Is this necessary because of the protected parent destructor? If so, add `CELER_DEFAULT_COPY_MOVE(GeantStepDiagnostic);` to satisfy the "rule of 5"